<?xml version="1.0"?>
<launch>

  <!-- PARAMETER SECTION -->
  <!--==================================================================-->
  <!-- SPAWN TAGS PARAMETERS DEFINITION -->
  <param name="tag_height" type="int" value="3" />
  <param name="tag_distance_rate" type="int" value="100" />
  <param name="tag_coordinate_file" type="str" value="$(find dist_project)/coordinates/tag_coordinates.csv" />
  <param name="tag_urdf_file" type="str" value="$(find dist_project)/urdf/tag.urdf" />

  <!-- SPAWN TARGET PARAMETERS DEFINITION -->
  <param name="target_coordinate_file" type="str" value="$(find dist_project)/coordinates/target_coordinates.csv" />
  <param name="target_sdf_file" type="str" value="$(find dist_project)/models/target/model.sdf" />

  <!-- SPAWN ROOM PARAMETERS DEFINITION -->
  <param name="room_sdf_file" type="str" value="$(find dist_project)/models/room/model.sdf" />

  <!-- ROBOTS PARAMETER DEFINITION-->
  <arg name="model" default="$(find dist_project)/urdf/unicycle.urdf.xacro" />
  <param name="robot_description" command="$(find xacro)/xacro $(arg model)" />
  <param name="robot_coordinate_file" type="str" value="$(find dist_project)/coordinates/robot_coordinates.csv" />

  <!-- UWB DISTANCE SIMULATOR NODE PARAMETERS -->
  <param name="initialization_time" type="int" value="5" />

  <!-- MOTION PLANNER NODE PARAMETERS -->
  <param name="motion_planner_rate" type="int" value="1" />

  <!-- KALMAN LOCALIZATION NODE PARAMETERS -->
  <param name="localization_rate" type="int" value="10" />

  <!-- TARGET ESTIMATOR NODE PARAMETERS -->
  <param name="target_estimator_rate" type="int" value="10" />

  <!-- NODES SECTION -->
  <!--==================================================================-->

  <!-- INITIALIZE GAZEBO NODE -->
  <node pkg="gazebo_ros" type="gazebo" name="gazebo" />
  <node pkg="dist_project" type="gazebo_init.py" name="gazebo_init" />

  <!-- INITIALIZE CAMERA POINT OF VIEW NODE -->
  <node pkg="rqt_image_view" type="rqt_image_view" name="camera_blob_viewer" />  

  <!-- INITIALIZE PLOT MANAGER NODE -->
  <node pkg="plotjuggler" type="plotjuggler" name="plotjuggler" />

</launch>
